src/command-helpers/network.js

```diff
@@ -0,0 +1,135 @@
+const path = require('path')
+const yaml = require('yaml')
+const { step, withSpinner } = require('../command-helpers/spinner')
+
+const updateSubgraphNetwork = async (toolbox, manifest, network, networksFile) =>
+  await withSpinner(
+    `Update sources network`,
+    `Failed to update sources network`,
+    `Warnings while updating sources network`,
+    async spinner => {
+      let allNetworks
+
+      try {
+        step(spinner, `Reading networks config`)
+        allNetworks = await toolbox.filesystem.read(networksFile, "json")
+      }  catch (error) {
+        throw new Error(error.message)
+      }
+
+      let networkConfig = allNetworks[network]
+
+      // Exit if the network passed with --network does not exits in networks.json
+      if(!networkConfig) {
+        throw new Error(`Network '${network}' was not found in '${networksFile}'`)
+      }
+
+      await toolbox.patching.update(manifest, content => {
+        let subgraph = yaml.parse(content)
+        let networkSources = Object.keys(networkConfig)
+        let subgraphSources = subgraph.dataSources.map(value => value.name);
+
+        // Update the dataSources network config
+        subgraph.dataSources = updateSources(
+          toolbox,
+          spinner,
+          network,
+          subgraph.dataSources,
+          networkSources,
+          networkConfig
+        )
+
+        // All data sources shoud be on the same network,
+        // so we have to update the network of all templates too.
+        if(subgraph.templates) {
+          subgraph.templates = updateTemplates(network, subgraph.templates)
+        }
+
+        let unsusedSources = networkSources.filter(x => !subgraphSources.includes(x))
+
+        unsusedSources.forEach(source => {
+          step(spinner, `dataSource '${source}' from '${networksFile}' not found in ${manifest}`)
+        })
+
+        let yaml_doc = new yaml.Document()
+        yaml_doc.contents = subgraph
+        return yaml_doc.toString()
+      })
+  })
+
+const initNetworksConfig = async(toolbox, directory) =>
+  await withSpinner(
+    `Initialize networks config`,
+    `Failed to initialize networks config`,
+    `Warnings while initializing networks config`,
+    async spinner => {
+      let subgraphStr = await toolbox.filesystem.read(path.join(directory, 'subgraph.yaml'))
+      let subgraph = yaml.parse(subgraphStr)
+
+      let networks = {}
+
+      subgraph.dataSources.forEach(source => {
+        let sourceConfig = {[source.name]: {}}
+
+        if (source.source.address) sourceConfig[source.name]["address"] = source.source.address
+        if (source.source.startBlock) sourceConfig[source.name]["startBlock"] = source.source.startBlock
+
+        networks[source.network] = sourceConfig
+      })
```

Comment:
This is more of a code style thing, you don't have to do it 🙂 

Personally I prefer avoiding mutability, and _I think_ this snippet does the same thing:

```javascript
const networks = subgraph.dataSources.reduce((acc, source) => ({
  ...acc,
  [source.network]: {
    [source.name]: {
      address: source.source.address,
      startBlock: source.source.startBlock,
    },
  },
}, {}))
```

Unless the `filesystem` API breaks if a field is there but the value is `underfined`.

Reply:
I tried with `reduce`, but i think it has an issue when there's one element in the array, and it returns the dataSource instead, even with the initial value `{}`

Reply:
I think we have to use Object.assing(), this way it works:
```javascript
const networks = subgraph.dataSources.reduce((acc, source) =>
  Object.assign(
    acc,
    {
      [source.network]: {
        [source.name]: {
          address: source.source.address,
          startBlock: source.source.startBlock,
        },
      },
    }
  )
, {})
```